src/perf/lib/ThroughputClient.cpp

```diff
@@ -0,0 +1,410 @@
+    if (!TryGetValue(argc, argv, "target", &Target)) {
+        WriteOutput("Must specify '-target' argument!\n");
+        return QUIC_STATUS_INVALID_PARAMETER;
+    }
+
+    uint16_t ip;
+    if (TryGetValue(argc, argv, "ip", &ip)) {
+        switch (ip) {
+        case 4: RemoteFamily = AF_INET; break;
+        case 6: RemoteFamily = AF_INET6; break;
+        }
+    }
+
+    TryGetValue(argc, argv, "length", &Length);
+    if (Length == 0) {
+        WriteOutput("Must specify a positive 'length'\n");
+        return QUIC_STATUS_INVALID_PARAMETER;
+    }
+
+    const char* localAddress;
+    if (TryGetValue(argc, argv, "bind", &localAddress)) {
+        if (!ConvertArgToAddress(localAddress, 0, &LocalIpAddr)) {
+            WriteOutput("Failed to decode IP address: '%s'!\nMust be *, a IPv4 or a IPv6 address.\n", localAddress);
+            return QUIC_STATUS_INVALID_PARAMETER;
+        }
+    }
+
+    // TODO: Core, since we need to support kernel mode
+    #if _WIN32 && !_KERNEL_MODE
+    uint16_t compartmentid;
+    if (TryGetValue(argc, argv, "comp",  &compartmentid)) {
+        NETIO_STATUS status;
+        if (!NETIO_SUCCESS(status = SetCurrentThreadCompartmentId(compartmentid))) {
+            WriteOutput("Failed to set compartment ID = %d: 0x%x\n", compartmentid, status);
+            return QUIC_STATUS_INVALID_PARAMETER;
+        } else {
+            WriteOutput("Running in Compartment %d\n", compartmentid);
+        }
+    }
+
+    uint8_t cpuCore;
+    if (TryGetValue(argc, argv, "core",  &cpuCore)) {
+        SetThreadAffinityMask(GetCurrentThread(), (DWORD_PTR)(1ull << cpuCore));
+    }
+#endif
+
+    TryGetValue(argc, argv, "sendbuf", &UseSendBuffer);
+
+    uint32_t ioSize = UseSendBuffer ? THROUGHPUT_DEFAULT_IO_SIZE_BUFFERED : THROUGHPUT_DEFAULT_IO_SIZE_NONBUFFERED;
+    TryGetValue(argc, argv, "iosize", &ioSize);
+    IoSize = ioSize;
+
+    uint32_t ioCount = UseSendBuffer ? THROGHTPUT_DEFAULT_SEND_COUNT_BUFFERED : THROUGHPUT_DEFAULT_SEND_COUNT_NONBUFFERED;
+    TryGetValue(argc, argv, "iocount", &ioCount);
+    IoCount = ioCount;
+
+    size_t Len = strlen(Target);
+    TargetData.reset(new char[Len + 1]);
+    QuicCopyMemory(TargetData.get(), Target, Len);
+    TargetData[Len] = '\0';
+
+    RawIoBuffer = new uint8_t[IoSize];
+
+    return QUIC_STATUS_SUCCESS;
+}
+
+struct ThroughputClient::SendRequest {
+    QUIC_SEND_FLAGS Flags {QUIC_SEND_FLAG_NONE};
+    QUIC_BUFFER QuicBuffer;
+    uint32_t IoSize;
+    SendRequest(
+        uint32_t Size
+        ) {
+        IoSize = Size;
+        QuicBuffer.Buffer = RawIoBuffer;
+        QuicBuffer.Length = 0;
+    }
+
+    void SetLength(
+        uint64_t BytesLeftToSend
+        ) {
+        if (BytesLeftToSend > IoSize) {
+            QuicBuffer.Length = IoSize;
+        } else {
+            Flags |= QUIC_SEND_FLAG_FIN;
+            QuicBuffer.Length = (uint32_t)BytesLeftToSend;
+        }
+    }
+};
+
+struct ShutdownWrapper {
+    HQUIC ConnHandle{nullptr};
```

Comment:
Like you do in SendRequest above, for all initializations, put a space between the variable and the {...} stuff.